### PR TITLE
Bugfix: Check for active controller before access getRequest()

### LINKED ISSUE #173: BUG: updateGrant() method throws an Exception when there is no active Controller
`ShareDraftContentFileShortcodeProviderExtension::updateGrant()`:

The following code does not consider that we might not have an active Controller when this method is called:
```php
$session = $session = Controller::curr()->getRequest()->getSession();
```

Result:
```
Call to a member function getRequest() on null
```

I haven't dug into the exact reasons for why this method might be called outside of the context of a Request, but it seems to happen often when using the Silverstripe Search Service module and reindexing items.

I will work on a PR shortly.

### PR DESCRIPTION
**Pull request by chrispenny**
Fixes #173

### CODE REVIEW

File: src/Extensions/ShareDraftContentFileShortcodeProviderExtension.php [hunk 1]
```diff
@@ -20,16 +20,27 @@ public function updateGrant(bool &$grant, ?File $record = null, ?array $args = n
...
         }
-        $session = $session = Controller::curr()->getRequest()->getSession();
+
+        $controller = Controller::curr();
+
+        if (!$controller) {
+            return;
+        }
+
+        $session = $controller->getRequest()->getSession();
```
Comment (ChloeSartorelli) [inline src/Extensions/ShareDraftContentFileShortcodeProviderExtension.php]:
```suggestion
        if (!Controller::has_curr()) {
            return;
        }

        $session = Controller::curr()->getRequest()->getSession();
```
`Controller::curr()` throws an error if there's no controller. We have to check if there's a current controller with `has_curr()`

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
Not super fond of all the added whitespace either but it's not harming anyone so I wont' ask you to remove it. There's no 'standard' for when to add new lines atm, so it's highly subjective.

File: src/Extensions/ShareDraftContentFileShortcodeProviderExtension.php (hunk 1 above)
Comment (chrispenny) [reply src/Extensions/ShareDraftContentFileShortcodeProviderExtension.php]: Good to know! I wasn't aware of that.

Regarding the whitespace: Spaces before/after control structures is a Slevomat standard. I think it's quite a popular standard - or at least, we use it for all of our bespoke projects.

**Review by GuySartorelli — APPROVED** [review]

**Comment by GuySartorelli** [comment]
Released as [2.7.2](https://github.com/silverstripe/silverstripe-sharedraftcontent/releases/tag/2.7.2)

**Pull request merged**